File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 12 - Apprentice/viewer demo C#/ApprenticeDemo/ApprenticeDemo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

 
namespace ApprenticeDemo
{
    public partial class Form1 : Form
    {
        string mFilename;

        public Form1()
        {
            InitializeComponent();
        }

        private void btnOpen_Click(object sender, EventArgs e)
        {
              mFilename = OpenFile();

            if (mFilename != "") 
                axInventorViewControl1.FileName = mFilename;
             
        }

        private string OpenFile()
        {

         string filename = "";

         System.Windows.Forms.OpenFileDialog ofDlg = new System.Windows.Forms.OpenFileDialog();

         string user = System.Windows.Forms.SystemInformation.UserName;

         ofDlg.Title = "Open Inventor File";
         ofDlg.InitialDirectory = "C:\\Documents and Settings\\" + user + "\\Desktop\\";

         //"Inventor Files (*.ipt)|*.ipt|Inventor Assemblies (*.iam)|*.iam|Inventor Drawings (*.idw)|*.idw"
         ofDlg.Filter = "Inventor files (*.ipt; *.iam; *.idw)|*.ipt;*.iam;*.idw";
         ofDlg.FilterIndex = 1;
         ofDlg.RestoreDirectory = true;

         if ((ofDlg.ShowDialog() == DialogResult.OK)) {
          filename = ofDlg.FileName;
         }

         return filename;

        }


    }
}

File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 12 - Apprentice/viewer demo Vb.Net/ApprenticeDemo/ApprenticeDemo/Form1.vb
Public Class Form1
    Dim mFilename As String
    Private Sub btnOpen_Click(sender As System.Object, e As System.EventArgs) Handles btnOpen.Click

        mFilename = OpenFile()

        If (mFilename <> "") Then
            AxInventorViewControl1.FileName = mFilename
        End If
    End Sub

    Private Function OpenFile() As String

        Dim filename As String = ""

        Dim ofDlg As System.Windows.Forms.OpenFileDialog = New System.Windows.Forms.OpenFileDialog()

        Dim user As String = System.Windows.Forms.SystemInformation.UserName

        ofDlg.Title = "Open Inventor File"
        ofDlg.InitialDirectory = "C:\Documents and Settings\" + user + "\Desktop\"

        '"Inventor Files (*.ipt)|*.ipt|Inventor Assemblies (*.iam)|*.iam|Inventor Drawings (*.idw)|*.idw"
        ofDlg.Filter = "Inventor files (*.ipt; *.iam; *.idw)|*.ipt;*.iam;*.idw"
        ofDlg.FilterIndex = 1
        ofDlg.RestoreDirectory = True

        If (ofDlg.ShowDialog() = DialogResult.OK) Then
            filename = ofDlg.FileName
        End If

        Return filename

    End Function
End Class
